src/core/server/integration_tests/status/routes/status.test.ts

```diff
@@ -110,25 +110,73 @@ describe('GET /api/status', () => {
   });
 
   describe('allowAnonymous: false', () => {
-    it('rejects requests with no credentials', async () => {
-      await setupServer({ allowAnonymous: false });
-      await supertest(httpSetup.server.listener).get('/api/status').expect(401);
+    describe('http response code', () => {
+      it('respond with a 200 when core.overall.status is available', async () => {
```

Comment:
I was lazy and didn't check if we might have an existing test like this... but it feels valuable to test that when core.overall.status is available, a critical plugin status can never cause /api/status to return non-200

Reply:
Yeah, the behavior didn't really change in this PR, but I was surprised that such test was not present. I added it in https://github.com/elastic/kibana/pull/159768/commits/b4fe7932486b1178dd976ac02d8289cc2fea4f5c